crashreporter/src/main/java/com/tealium/crashreporter/Crash.kt

```diff
@@ -0,0 +1,162 @@
+
+final class Crash (Thread thread, Throwable ex) {
+    const val KEY_STACK_CLASS_NAME = "className"
+    const val KEY_STACK_FILENAME = "fileName"
+    const val KEY_STACK_LINE_NUMBER = "lineNumber"
+    const val KEY_STACK_METHOD_NAME = "methodName"
+    const val KEY_THREAD_STATE = "state"
+    const val KEY_THREAD_NUMBER = "threadNumber"
+    const val KEY_THREAD_ID = "threadId"
+    const val KEY_THREAD_PRIORITY = "priority"
+    const val KEY_THREAD_STACK = "stack"
+
+    private val mException: Throwable
+    private val mThread: java.lang.Thread
+
+    private val mExceptionCause: String
+    private val mExceptionName: String
+    private var mUuid: String
+    private var mThreadState: String
+    private var mThreadNumber: String
+    private var mThreadPriority: String
+
+    init {
+        mException = ex
+        mThread = thread
+    }
+
+
+    fun getCause(): String {
```

Comment:
I believe can do without many of these getters. Checkout Kotlin [Getters/Setters](https://kotlinlang.org/docs/reference/properties.html#getters-and-setters)